go/boilerplate/future.go

```diff
@@ -0,0 +1,45 @@
+package boilerplate
```

Comment:
The contents of this file were selectively copied from https://github.com/estuary/flow/blob/master/go/protocols/flow/re_exports.go. `OpFuture` is only used in the Flow repo for the sqlite materialization there & the protobuf-based materialization transactor, both of which should be going away. It makes more sense to have this here where it will be used directly by materialization connectors, and eventually it can be removed entirely from the Flow repo when all materialization connectors have converted to the JSON protocol.